src/promnesia/common.py

```diff
@@ -40,7 +40,16 @@ def file(cls, path: PathIsh, line: Optional[int]=None, relative_to: Optional[Pat
         ll = '' if line is None else f':{line}'
         # todo loc should be url encoded? dunno.
         # or use line=? eh. I don't know. Just ask in issues.
-        handler = _detect_mime_handler()
+
+        # todo: handler has to be overridable by config. This is needed for docker, but also for a "as a service" install, where the sources would be available on some remote webserver
```

Comment:
hm, not sure what you mean here? If you install https://github.com/karlicoss/open-in-editor in Docker it's definitely not going to work either way? 
Or do you mean that paths mounted in docker won't work on the host system? I generally just use the same paths when using Docker, makes everything much easier.

Reply:
Right, i mean that paths will be relative to the docker root. Using the same paths sounds like a nice trick, but still. And, running promnesia on a server, i want to prepend some "http://my-server/path/to/data/", *and* ensure that ":line" is not appended. Well, this makes me think that for a multi-user setup, we could eventually make it a setting *in the client*. But that's a long way ahead

Reply:
`eval()`'ed hooks sound like a good way to get some rudimentary customizability going, yeah.
But here i'm talking a usecase of getting my clients and coworkers and friends (yes, all of them) to install this extension. The most i can ask them is to `install it, go to settings and set the server to <url> and click save`, or other similarly basic sequence of actions. It could possibly be `install it, go to settings and insert <url> into "hooks list" or something like that. 

But it's the server that knows where it stores data and how people can access them, so i don't see how it shouldnt' be the server that gives this info to the client. Not necessarily by composing a fixed "href" as it does now. Sooner or later the client will need to be smarter, so we may as well start giving it data in a more semantic way: pass is the path, line number, and a suggested pattern to put this together, maybe?

Maybe you'd rather like to see these responsibilities split between the indexer and the "server"? That'd make sense to me.

Reply:
> setting to prepend the paths is a bit too specific anyway.

yeah.. but hey, it will get the job done

Reply:
whatever design that will be aligned with your mental model though!

Reply:
> not quite the robust solution i'm looking for

Yeah, agree it's kind of unsatisfactory. Although in general I've had much less grief with Docker since I started doing this, but that's just me :)

> yeah.. but hey, it will get the job done

Yep, generally I'm all for it, but I'm trying to be cautions with adding items to the config, because it's kind of a promise to keep it backwards compatible.

-----
Otherwise, agreed, makes sense! Especially the point that it doesn't seem like frontend's responsibility.

I guess a way that would make more sense without any extra code changes is running an sqlite query right after indexing to 'fix' the paths? Still kinda hacky, but you only need to do this once and it's very flexible. But my concern with that would be that eventually I want to support [realtime indexing](https://github.com/karlicoss/promnesia/issues/20), and this wouldn't quite work.

Another, nicer but stil not too specific option would be a config hook, through which all visits pass before getting written in the database? E.g. if I understood your usecase, it would look like:

```python
# in config.py

def hook(v: Visit): # this will have a nicer name?
    href = v.locator.href or ""
    if "editor://" not in href:
          return # it's fine, not need to patch
    href = href.replace("editor://", "http://your-serlver")
    href = re.replace(r":.*", "", href) # chop off line number
    v.locator.href = href
    return v
```

The upside is that it will be quite flexible and potentially support usecases we couldn't even anticipate. I even wouldn't mind merging specific 'hooks' in the repository, so they could be imported and used in `config.py`. What do you think?